dScripts/CppScripts.cpp

```diff
@@ -327,668 +327,382 @@
 #include "LupGenericInteract.h"
 #include "WblRobotCitizen.h"
 
+#include <map>
+#include <string>
+#include <functional>
+
 namespace {
-	// This is in the translation unit instead of the header to prevent wierd linker errors
-	InvalidScript* const InvalidToReturn = new InvalidScript();
-	std::map<std::string, CppScripts::Script*> m_Scripts;
+    // This is in the translation unit instead of the header to prevent weird linker errors
+    InvalidScript InvalidToReturn;
+    std::map<std::string, CppScripts::Script*> m_Scripts;
+    std::map<std::string, std::function<CppScripts::Script* ()>> test = {
```

Comment:
g_Scripts would be a better name for this map

Reply:
Made change